app/controllers/lead_providers/confirm_schools_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class ConfirmSchoolsController < ApplicationController
+    before_action :load_form
+
+    def show
+      @schools = School.where(id: @confirm_schools_form.school_ids).order(:name).includes(:local_authority)
+      @delivery_partner = DeliveryPartner.find(@confirm_schools_form.delivery_partner_id)
```

Comment:
How do you feel about moving this logic into the form? Having a `schools` and `delivery_partner` method that returns these?

Reply:
So yes, it seems to be repeating now. The only thing I'd be very here though is additional coupling. As written, the form is completely oblivious to what uuids represent, and as such is pretty much just a silly data structure encapsulating absolutely no behaviour, leaving it to a controller to actually interpret that data. But, the benefit of this is pretty low, so I think I'll just make it consistent with other forms

Reply:
Should make it testable too, right?

Reply:
I personally prefer taking logic out of the controller where possible. How does that fit with your design patterns?

Reply:
Ideally I'd move pretty much all the logic out of controller and split it onto domain-specific dervice objects and web-specific details. I think we'd need to make a decision here though - form is supposed to be a simple data structure for validating input. Once we start adding more business logic and business transformations to it than they are no longer forms, but domain processes (which seems like you are shifting towards as well!)

Reply:
Forms can be whatever we want them to be. 

In my mind, basic Rails thinks models should do a lot. They should validate data. They should be used to create forms in views. Among other things, like creating / updating entries in our database, or so that we can access some fields on the model in views from a single `@` variable, rather than many.

I think the reason we use forms is that we have use cases which go a bit outside what Rails wants - either validation that is not meant to be on the model, or manipulating data that doesn't belong on a single model. And so our forms are really a layer of data manipulation that's more flexible than models. That's why in a lot of places we have methods like `save` (with a transaction), or loading data in the form - because really, they are replicating what Rails models are capable of.

I would be weary of restricting our idea of the form to something more narrow, precisely because it will raise the need for something that will load the data, or save things to commonise what controllers are doing. Kind of like what's happening here. 

Also, in several places we still use Rails models instead of forms - whether we use a form is kind of dictated by whether our use case is possible to achieve using models or not. In that way, models and forms are kind of interchangeable when it comes to their place in their capabilities - and truth be told I don't really see the need to change it.

Reply:
I guess the main thing is 'what do we call them, and is `form` a bad name?' :D 

Maybe? `form_with @confirm_schools_domain_object` doesn't roll off the tongue.

Reply:
@kicferk1 Absolutely, Rails way is wrong on so many leveles. Which is why I always suggest a trailblazer-style coding: https://github.com/trailblazer/trailblazer. There is a concept there called `Operation` which should contain all the business logic (so - empty controllers). `Forms` are only responsible for validations and they are used within the operations. I'd absolutely love to write few spikes using these concepts (however I would not use trailblazer implementations as they are also wrong in quite a feww places :P)